WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetails.swift

```diff
@@ -8,7 +8,7 @@ struct ShippingLabelPackageDetails: View {
 
     /// Completion callback
     ///
-    typealias Completion = (_ selectedPackages: [ShippingLabelPackageAttributes]) -> Void
+    typealias Completion = (_ packagesResponse: ShippingLabelPackagesResponse?, _ selectedPackages: [ShippingLabelPackageAttributes]) -> Void
```

Comment:
I think just updating package response on completion (after selection a package) is not sufficient. There's a case where a user adds new package but then navigates out of the Package Details screen without select any package - here the package response is not updated. 

I'd suggest creating a new callback block just for updating package response, inject into the view model and call it at the end of `handleNewPackage` method. That way we can be sure that the package list is updated whenever a new package is added.

Reply:
It's also nice if we can add both blocks to the view model's initializer instead of keeping one in the view. Tapping Done will then trigger a method in the view model that calls the completion block.

Reply:
Good catch! I didn't consider that case before. Thank you for the really clear suggestions; that makes a lot of sense. I made those suggested changes in 08a9e47, including putting both blocks in the view model.